pkg/minikube/kubeconfig/update.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeconfig
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/docker/machine/libmachine/host"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	cfg "k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/minikube/exit"
+)
+
+// Update sets up kubeconfig to be used by kubectl
+func Update(h *host.Host, c *cfg.Config) *Setup {
+	addr, err := h.Driver.GetURL()
```

Comment:
Instead of accepting host.Host, could this could accept a proper URL for ClusterServerAddress?

I feel like making this library depend on a large library like host makes the interface muddy and forces it to handle implementation details it shouldn't need to know. It would make testing nicer too.